src/fromager/vendor_rust.py

```diff
@@ -84,24 +87,56 @@ def _cargo_config(project_dir: pathlib.Path):
         toml.dump(cfg, f)
 
 
+def _should_vendor_rust(req: Requirement, project_dir: pathlib.Path) -> bool:
+    """Detect if project has build requirement on Rust
+
+    Detects setuptools-rust and maturin.
+    """
+    # setuptools-rust and maturin-based projects have a pyproject.toml
+    pyproject_toml = project_dir / "pyproject.toml"
+    if not pyproject_toml.is_file():
+        logger.debug(f"{req.name}: has no pyproject.toml")
+        return False
+
+    with pyproject_toml.open(encoding="utf-8") as f:
+        pyproject = toml.load(f)
+
+    try:
+        build_requires = pyproject["build-system"]["requires"]
+    except KeyError:
+        logger.debug(f"{req.name}: no [build-system]requires in pypyprojet.toml")
+        return False
```

Comment:
A lot of the logic above reproduces https://github.com/python-wheel-build/fromager/blob/main/src/fromager/dependencies.py#L222. Maybe it makes sense to have it inline here anyway, to include the debug log messages?

Reply:
Valid point. I have rewritten the code to use the existing helpers.